modules/apps/roles/roles-admin-web/src/main/resources/META-INF/resources/js/EditRolePermissionsNavigation.es.js

```diff
@@ -0,0 +1,175 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClayInput} from '@clayui/form';
+import {ClayVerticalNav} from '@clayui/nav';
+import {useDebounce} from '@clayui/shared';
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+
+export default function EditRolePermissionsNavigation({
+	items,
+	portletNamespace,
+}) {
+	const [activeItemId, setActiveItemId] = useState();
+	const [expandedKeys, setExpandedKeys] = useState(new Set([]));
+	const [filterQuery, setFilterQuery] = useState('');
+
+	const debouncedFilterQuery = useDebounce(filterQuery, 200);
+
+	const handleItemClick = useCallback(
+		(item, portletNamespace) => {
+			const method = `${portletNamespace}loadContent`;
+
+			window[method](item.resourceURL);
+
+			setActiveItemId(item.id);
+		},
+		[setActiveItemId]
+	);
+
+	const processItems = useCallback(
+		(items) => {
+			const processedExpandedKeys = new Set([]);
+			const processedItems = [];
+			let hasActiveChild = false;
+
+			items.forEach((item) => {
+				const {
+					className,
+					id,
+					items: childItems,
+					label,
+					resourceURL,
+				} = item;
+				const normalizedFilterQuery = debouncedFilterQuery
+					.trim()
+					.toLowerCase();
+
+				const hasFilterQuery = normalizedFilterQuery !== '';
+
+				const showItem =
+					!hasFilterQuery ||
+					item.items ||
+					item.ignoreFilter ||
+					label.toLowerCase().includes(normalizedFilterQuery);
+
+				if (showItem) {
+					const processedItem = {
+						id,
+						label,
+					};
+
+					const active = activeItemId
+						? activeItemId === id
+						: item.active;
+
+					if (active) {
+						hasActiveChild = true;
+
+						processedItem.active = true;
+					}
+
+					if (className) {
+						processedItem.className = className;
+					}
+
+					if (resourceURL) {
+						processedItem.onClick = () =>
+							handleItemClick(item, portletNamespace);
+					}
```

Comment:
These states you can add directly in the component rendering instead of adding in the filter.

```jsx
{(item) => (
  <ClayVerticalNav.Item {...item} onClick={item.resourceURL ?  () => handleItemClick(item, portletNamespace) : undefined}>
    {item.label}
  </ClayVerticalNav.Item>
)}
```

`portletNamespace` also wouldn't even need to be passed to the function could get it via hosting since it's in the same scope.

Reply:
Updated accordingly... Thanks @matuzalemsteles !